frontend/src/employee-frontend/components/application-page/ApplicationEditView.tsx

```diff
@@ -718,9 +725,42 @@ export default React.memo(function ApplicationEditView({
               <InputWarning text={i18n.application.preferences.unitMismatch} />
             ) : null}
             {preferredUnits.map((unit, i) => (
-              <HorizontalContainer key={unit.id}>
+              <PreferredUnitGridContainer key={unit.id}>
                 <Link to={`/units/${unit.id}`}>{`${i + 1}. ${unit.name}`}</Link>
-                <Gap size="s" horizontal />
+                <Button
+                  appearance="inline"
+                  icon={faArrowUp}
+                  text={i18n.documentTemplates.templateEditor.moveUp}
+                  onClick={() =>
+                    setApplication(
```

Comment:
Tämä olisi helpommin luettava jos käyttäisi helper funktiota array slicauksen sijaan. Esim:
```ts
function swapElements<T>(array: T[], index1: number, index2: number): T[] {
  const result = [...array]
  const temp = result[index1]
  result[index1] = result[index2] 
  result[index2] = temp
  return result
}
käyttö tässä:
```

```suggestion
                    setApplication(set('form.preferences.preferredUnits', swapElements(preferredUnits, i, i - 1)))
```
ja vastaavasti alaspäin nappulassa
```tsx
onClick={() => setApplication(set('form.preferences.preferredUnits', swapElements(preferredUnits, i, i + 1)))
```